src/main/scala/nodes/learning/KernelMatrix.scala

```diff
@@ -0,0 +1,78 @@
+package nodes.learning
+
+import breeze.linalg._
+import edu.berkeley.cs.amplab.mlmatrix.{RowPartition, NormalEquations, BlockCoordinateDescent, RowPartitionedMatrix}
+import nodes.stats.{StandardScalerModel, StandardScaler}
+import nodes.util.{VectorSplitter, Identity}
+import org.apache.spark.broadcast.Broadcast
+import org.apache.spark.rdd.RDD
+import scala.collection.mutable.HashMap
+import utils.{MatrixUtils, Stats}
+import workflow.{Transformer, LabelEstimator}
+
+/* Defines a distributed KernelMatrix
+ * Note this is a lazy matrix, 
+ */
+
+trait KernelMatrix {
+
+  /* Return a n x idxs.size sub matrix (every entry in RDD is a row)
+   * @param idxs the column indexes of your kernel matrix you want
+   * */
+  def apply(idxs: Seq[Int]):RDD[DenseMatrix[Double]]
+
+  def apply(idxs1: Seq[Int], idxs2: Seq[Int]): RDD[DenseMatrix[Double]]
```

Comment:
I assume this is subsampling rows as well as columns? Would be nice to change cariable names to be like `rowIdxs` and `colIdxs`, to make it more explicit.